### PR TITLE
basics of js 

### PR DESCRIPTION
**Pull request by Krishnakant1101**

### CODE REVIEW

File: Topic-Array_methods/example.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+const fruits2 = ["Banana", "Orange", "Apple", "Mango"];
+let fruit = fruits2.at(2);
+console.log(fruit);
+
+/*.............................................................................................................*/
+
+const fruits3 = ["Banana", "Orange", "Apple", "Mango"];
+let s2 = fruits3.join(" * ");
+console.log(s2);
+
```
Comment (Ritikthoughtwin) [inline Topic-Array_methods/example.js:31]: Naming conventions and declarations should be informative.

File: Topic-Array_methods/example.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+let position1 = fruits12.lastIndexOf("Apple") 
+
+
+/*......................................................................................................................................*/
+
+
+const fruits13 = ["Banana", "Orange", "Apple", "Mango"];
+fruits13.includes("Mango"); // is true 
+
+
```
Comment (Ritikthoughtwin) [inline Topic-Array_methods/example.js:163]: Create only one variable declaration if you are using it multiple times.

File: Topic-Async_await/examples.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+myFunction().catch((error) => {
+    console.log("error : ", error);
+})
+
+/*..............................................................................................................*/
+
+let value = await fetch("");/* await keyword is used to wait or hold 
+                               the execution of async function for any specific  part 
+                               of the code */
+
```
Comment (Ritikthoughtwin) [inline Topic-Async_await/examples.js:49]: Also, create examples with arrow functions.

File: Topic-Callback_functions/examples.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+function mainFunction(callback) {
+  console.log("Performing operation...");
+  numbers.forEach(callback);
+}
+function callbackFunction(number) {
+  console.log("Result: " + number);
+}
+
+mainFunction(callbackFunction);
+
```
Comment (Ritikthoughtwin) [inline Topic-Callback_functions/examples.js:38]: Also, create examples with arrow functions.

File: Topic-Closure/Example_01.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+function parent() {
+    const name = "krishnakant Thakur";
+    function child() {
+        console.log("Name : ", name);
+    }
+    child();
+}
+
+parent();
```
Comment (Ritikthoughtwin) [inline Topic-Closure/Example_01.js]: Also, create examples with arrow functions.

File: Topic-Closure/Example_03.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+function outer(x){
+
+    function inner(y){
+        console.log("X : ",x," and Y : ",y);
+    }
+    return inner;
+
+}
+
```
Comment (Ritikthoughtwin) [inline Topic-Closure/Example_03.js:9]: Use informative function arguments.

File: Topic-Hoisting/function_hoisting.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    console.log("Hello");
+}
+
+
+print_hello();
+//use let,var or const
+let print_hello=function (){   //will not work
+    console.log("Hello");
+}
+
```
Comment (Ritikthoughtwin) [inline Topic-Hoisting/function_hoisting.js:13]: Fix naming conventions

**Comment by tw-vineet** [comment]
Keep all task folders in one folder following with your name.

File: Topic-Promises/Example.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  })
+
+  myPromise.catch((res)=>{     //.catch()
+    console.log(res);
+  })
+
+  /********************************************************************************************* */
+  
+  //Another Example 
+
```
Comment (Ritikthoughtwin) [inline Topic-Promises/Example.js]: Fix naming conventions.

File: Topic-Promises/Example.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  let promise = new Promise(function (resolve, reject) {
+    const x = 45;
+    const y = 45;
+    if (x === y) {
+        resolve();
+    } else {
+        reject();
+    }
+});
+
```
Comment (Ritikthoughtwin) [inline Topic-Promises/Example.js:133]: Seems you are copy-pasting code from outsource.

File: Topic-String_methods/examples.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+let text0 = "Hello world!";
+let result = text0.repeat(4);
+
+/*....................................................................................................................................*/
+
+
+let text_ = "Please visit Microsoft!";
+let newText = text_.replace("Microsoft", "W3Schools");
+
```
Comment (Ritikthoughtwin) [inline Topic-String_methods/examples.js:111]: Follow naming conventions.

**Comment by tw-vineet** [comment]
Add definitions based on all topics

File: Topic-Array_methods/example.js (hunk 2 above)
Comment (Krishnakant1101) [reply Topic-Array_methods/example.js:163]: yes, i have make a changes in my Array.js file  by creating  a global variable which is using by every example

File: Topic-Callback_functions/examples.js (hunk 4 above)
Comment (Krishnakant1101) [reply Topic-Callback_functions/examples.js:38]: i done it

File: Topic-Closure/Example_03.js (hunk 6 above)
Comment (Krishnakant1101) [reply Topic-Closure/Example_03.js:9]: i understood

File: Topic-Hoisting/function_hoisting.js (hunk 7 above)
Comment (Krishnakant1101) [reply Topic-Hoisting/function_hoisting.js:13]: ok

File: Topic-Promises/Example.js (hunk 8 above)
Comment (Krishnakant1101) [reply Topic-Promises/Example.js]: done

File: Topic-Promises/Example.js (hunk 9 above)
Comment (Krishnakant1101) [reply Topic-Promises/Example.js:133]: yes some of them yes i did. i will impove it

File: Topic-String_methods/examples.js (hunk 10 above)
Comment (Krishnakant1101) [reply Topic-String_methods/examples.js:111]: i make changes

File: Topic-Array_methods/example.js (hunk 1 above)
Comment (Krishnakant1101) [reply Topic-Array_methods/example.js:31]: i understood

**Comment by Krishnakant1101** [comment]
I updated my all files according to your comments. You can review it again

File: Topic-Array_methods/example.js [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+const fruits = ["Banana", "Orange", "Apple", "Mango"];
+/*.............................................................................................................................*/
+
+let size = fruits.length; //it should print length of an array
+console.log(size);        //output : 4
+
+/*............................................................................................................*/
+
+
+let s=fruits.toString();     // it should convert array into string (a comma seperated string)
```
Comment (Ritikthoughtwin) [inline Topic-Array_methods/example.js]: Fix naming conventions

File: Topic-Array_methods/example.js [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+
+
+let fruit = fruits.at(2);     // fruits.at() function shows the value stored in given index
+console.log(fruit);           // output : Apple
+
+/*.............................................................................................................*/
+
+
+let s2 = fruits.join(" * ");       /*  
+                                        .join ("*") array function returns the string
```
Comment (Ritikthoughtwin) [inline Topic-Array_methods/example.js]: Fix naming conventions

File: Topic-Array_methods/example.js [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+                          // and it returns the removed value
+
+/*.................................................................................................*/
+                                      // The unshift methods is used to add value from front side of an array and it also makes changes in actual array 
+fruits7.unshift("Lemon");
+                                      //and it returns  the current length of an array after changes
+/*............................................................................................................*/
+
+const num_01 = [1,2,3,4,5,6,7,8,9,10];
+const num_01 = [11,12,13,14,15];
```
Comment (Ritikthoughtwin) [inline Topic-Array_methods/example.js]: Fix naming conventions.

File: Topic-Array_methods/example.js [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+
+// with numbers 
+
+   const n = [40, 100, 1, 5, 25, 10];
+   n.sort(function(a, b){return a - b});   //for ascending order
+
+
+   n.sort(function(a, b){return b - a});     //for descending order
+
+
```
Comment (Ritikthoughtwin) [inline Topic-Array_methods/example.js:225]: Add proper naming conventions

File: Topic-Async_await/examples.js [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        const response = await fetch("https://jsonplaceholder.typicode.com/posts");
+        const data = await response.json();  // Wait for the response to be parsed into JSON
+        console.log("Data fetched:", data);
+    } catch (error) {
+        console.log("Fetch Error:", error);  // Handles any error during the fetch
+    }
+}
+
+fetchDataFromAPI();
+
```
Comment (Ritikthoughtwin) [inline Topic-Async_await/examples.js:48]: Add example for arrow functions as well.

File: Topic-Closure/Example_02.js [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    value=34;
+    function inner(){
+        return value;
+    }
+    return inner;
+}
+
+const x=outer();
+const y=x();
+
```
Comment (Ritikthoughtwin) [inline Topic-Closure/Example_02.js:11]: Rename naming convention

File: Topic-Array_methods/example.js (hunk 11 above)
Comment (Krishnakant1101) [reply Topic-Array_methods/example.js]: done

File: Topic-Array_methods/example.js (hunk 12 above)
Comment (Krishnakant1101) [reply Topic-Array_methods/example.js]: done

File: Topic-Array_methods/example.js (hunk 14 above)
Comment (Krishnakant1101) [reply Topic-Array_methods/example.js:225]: done

File: Topic-Async_await/examples.js (hunk 15 above)
Comment (Krishnakant1101) [reply Topic-Async_await/examples.js:48]: done

File: Topic-Closure/Example_02.js (hunk 16 above)
Comment (Krishnakant1101) [reply Topic-Closure/Example_02.js:11]: done

**Comment by Krishnakant1101** [comment]
All the comments are resolved please review again

**Review by Ritikthoughtwin — APPROVED** [review]

**Pull request merged**